### PR TITLE
fix pca

### PR DESCRIPTION
**Pull request by valer1435**
This is a 🐛 bug fix. 

  Fixes [#4567](https://github.com/aimclub/FEDOT/issues/1266)

### CODE REVIEW

**Comment by valer1435** [comment]
@open-code-helper

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

File: fedot/core/operations/evaluation/operation_implementations/data_operations/sklearn_transformations.py [hunk 1]
```diff
@@ -23,6 +24,8 @@ class ComponentAnalysisImplementation(DataOperationImplementation):
         params: OpearationParameters with the arguments
     """
 
+    MIN_THRESHOLD_TS = 7
```
Comment (Lopa10ko) [inline fedot/core/operations/evaluation/operation_implementations/data_operations/sklearn_transformations.py:27]: хотелось бы магические числа протягивать из `fedot/core/constants.py`

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper

**Comment by valer1435** [comment]
@open-code-helper

**Comment by valer1435** [comment]
@open-code-helper

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

**Comment by valer1435** [comment]
@open-code-helper run

File: fedot/core/operations/evaluation/operation_implementations/data_operations/sklearn_transformations.py (hunk 1 above)
Comment (valer1435) [reply fedot/core/operations/evaluation/operation_implementations/data_operations/sklearn_transformations.py]: [Переместил](https://github.com/aimclub/FEDOT/pull/1267/commits/368c2fba016b9b63d53bf16cba7a771b348404ba)

**Review by Lopa10ko — APPROVED** [review]

File: fedot/core/operations/evaluation/operation_implementations/data_operations/sklearn_transformations.py [hunk 2]
```diff
@@ -80,12 +85,15 @@ def check_and_correct_params(self):
             # Check that n_samples correctly map with n_features
             if self.number_of_samples < self.number_of_features:
                 self.params.update(n_components=0.5)
+        if is_ts_data and (n_components * self.number_of_features) < PCA_MIN_THRESHOLD_TS:
+            self.params.update(n_components=PCA_MIN_THRESHOLD_TS / self.number_of_features)
```
Comment (ChrisLisbon) [inline fedot/core/operations/evaluation/operation_implementations/data_operations/sklearn_transformations.py:89]: Не будет проблем с нецелым числом n_components?

**Review by ChrisLisbon — APPROVED** [review]

File: fedot/core/operations/evaluation/operation_implementations/data_operations/sklearn_transformations.py (hunk 2 above)
Comment (valer1435) [reply fedot/core/operations/evaluation/operation_implementations/data_operations/sklearn_transformations.py:89]: If 0 < n_components < 1 and svd_solver == 'full', select the number of components such that the amount of variance that needs to be explained is greater than the percentage specified by n_components.

У нас 0 < n_components < 1

**Pull request merged**